### ISSUE TITLE
Daemon freezing.

### ISSUE THREAD

**Issue by p3yot3** [goldcoin/goldcoin#11]
I'm seeing my GLD daemon freezing up constantly in my logs:

"2017-01-25 21:44:46.074803 > ########################################
2017-01-25 21:44:46.074845 P2Pool: 18643 shares in chain (15385 verified/18647 total) Peers: 2 (1 incoming)
2017-01-25 21:44:46.074875  Local: 1047kH/s in last 10.0 minutes Local dead on arrival: ~5.1% (3-8%) Expected time to share: 12.3 minutes
2017-01-25 21:44:46.074905  Shares: 2 (1 orphan, 0 dead) Stale rate: ~50.0% (9-91%) Efficiency: ~99.9% (18-181%) Current payout: (0.0005)=0.0005 GLD
2017-01-25 21:44:46.074936  Pool: 508MH/s Stale rate: 49.9% Expected time to block: 5.0 minutes
2017-01-25 21:44:49.080877 > ########################################
2017-01-25 21:44:49.081007 > >>> Warning: LOST CONTACT WITH BITCOIND for 2.9 minutes! Check that it isn't frozen or dead!
2017-01-25 21:44:49.081050 > ########################################
2017-01-25 21:44:49.081087 P2Pool: 18643 shares in chain (15385 verified/18647 total) Peers: 2 (1 incoming)
2017-01-25 21:44:49.081117  Local: 1040kH/s in last 10.0 minutes Local dead on arrival: ~5.2% (3-8%) Expected time to share: 12.3 minutes
2017-01-25 21:44:49.081143  Shares: 2 (1 orphan, 0 dead) Stale rate: ~50.0% (9-91%) Efficiency: ~99.9% (18-181%) Current payout: (0.0005)=0.0005 GLD
2017-01-25 21:44:49.081173  Pool: 508MH/s Stale rate: 49.9% Expected time to block: 5.0 minutes
2017-01-25 21:44:52.090309 > ########################################
2017-01-25 21:44:52.090426 > >>> Warning: LOST CONTACT WITH BITCOIND for 2.9 minutes! Check that it isn't frozen or dead!
2017-01-25 21:44:52.090464 > ########################################
2017-01-25 21:44:52.090500 P2Pool: 18643 shares in chain (15385 verified/18647 total) Peers: 2 (1 incoming)
2017-01-25 21:44:52.090524  Local: 1030kH/s in last 10.0 minutes Local dead on arrival: ~5.2% (3-8%) Expected time to share: 12.5 minutes
2017-01-25 21:44:52.090550  Shares: 2 (1 orphan, 0 dead) Stale rate: ~50.0% (9-91%) Efficiency: ~99.9% (18-181%) Current payout: (0.0005)=0.0005 GLD
2017-01-25 21:44:52.090579  Pool: 508MH/s Stale rate: 49.9% Expected time to block: 5.0 minutes
2017-01-25 21:44:52.481304 > Error getting work from bitcoind:
2017-01-25 21:44:52.481480 > Traceback (most recent call last):
2017-01-25 21:44:52.481626 > Failure: twisted.internet.defer.TimeoutError: Getting http://127.0.0.1:8122/ took longer than 30 seconds.
2017-01-25 21:44:55.099654 > ########################################
2017-01-25 21:44:55.099868 > >>> Warning: LOST CONTACT WITH BITCOIND for 3.0 minutes! Check that it isn't frozen or dead!
2017-01-25 21:44:55.099965 > ########################################

This happens both randomly & whenever a block is found, so +400 times a day for up to 10 minutes per time!

Running the latest version on Ubuntu 14.04 64bit

**Comment by akumaburn** [comment]
There's no reason for the RPC to freeze up that i can think of. Something must be wrong on your end.

Even the issue Bittrex is having is completely unrelated to this.

**Comment by p3yot3** [comment]
Nothing wrong my end, I just had a quick read through the BCT thread for GLD & it seems that all p2pool nodes are suffering from the same issue. I turned it off anyway, it's unusable with p2pool.

I wasn't aware Bittrex also had problems, might be worth double checking that the problems aren't related.

**Comment by IconFirm** [comment]
I can confirm this error too - I've been battling with this problem for a week now, here are my daemon logs:

ProcessBlock: ACCEPTED
trying connection 80.86.85.65:8121 lastseen=384934.3hrs
CreateNewBlock(): total size 2775
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c0832fd  000000000832fd00000000000000000000000000000000000000000000000000
After:  1c08217e  0000000008217e00000000000000000000000000000000000000000000000000
ThreadRPCServer method=getblock
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 41.230.226.144:8121 lastseen=807.7hrs
connect() failed after select(): Connection refused
trying connection 85.140.3.210:8121 lastseen=384934.3hrs
Flushing wallet.dat
Flushed wallet.dat 12ms
connection timeout
trying connection 189.15.214.246:8121 lastseen=214.3hrs
connection timeout
trying connection 88.206.141.110:8121 lastseen=384934.3hrs
connect() failed after select(): Connection refused
trying connection 189.15.214.246:8121 lastseen=214.3hrs
connect() failed after select(): No route to host
trying connection 88.206.186.58:8121 lastseen=384934.3hrs
connect() failed after select(): Connection refused
connected 88.206.186.58:8121
send version message: version 60018, blocks=563496, us=0.0.0.0:0, them=88.206.186.58:8121, peer=88.206.186.58:8121
trying connection 50.161.251.31:8121 lastseen=415.2hrs
trying connection 91.121.6.41:8121 lastseen=384934.3hrs
connected 91.121.6.41:8121
send version message: version 60018, blocks=563496, us=0.0.0.0:0, them=91.121.6.41:8121, peer=91.121.6.41:8121
trying connection 91.134.233.254:8121 lastseen=384934.3hrs
socket closed
disconnecting node 91.121.6.41:8121
connection timeout
getheaders 563496 to 3a3a8a5c1d8a38635ca7
trying connection 100.80.47.19:8121 lastseen=798.6hrs
connection timeout
trying connection 91.201.245.200:8121 lastseen=384934.3hrs
connection timeout
trying connection 176.60.202.206:8121 lastseen=786.2hrs
connection timeout
trying connection 92.3.45.1:8121 lastseen=384934.3hrs
connection timeout
trying connection 191.179.157.134:8121 lastseen=589.6hrs
connection timeout
connect() failed after select(): Connection refused
trying connection 93.139.160.78:8121 lastseen=384934.3hrs
trying connection 84.246.202.191:8121 lastseen=1108.2hrs
connect() failed after select(): Connection refused
connect() failed after select(): No route to host
trying connection 78.70.227.24:8121 lastseen=145.4hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 168.194.79.186:8121 lastseen=805.5hrs
connection timeout
trying connection 92.253.185.128:8121 lastseen=407.6hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 182.70.220.178:8121 lastseen=310.2hrs
connection timeout
trying connection 86.121.68.233:8121 lastseen=26117.5hrs
connection timeout
trying connection 178.34.59.59:8121 lastseen=231.2hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 197.211.63.54:8121 lastseen=600.3hrs
connection timeout
trying connection 120.188.81.78:8121 lastseen=209.5hrs
socket no message in first 60 seconds, 0 1
disconnecting node 88.206.186.58:8121
connection timeout
trying connection 73.139.37.255:8121 lastseen=149.0hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 87.5.49.142:8121 lastseen=22096.0hrs
Flushed 12376 addresses to peers.dat  47ms
connection timeout
trying connection 95.28.51.179:8121 lastseen=1145.2hrs
connection timeout
trying connection 120.188.81.78:8121 lastseen=209.5hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 95.25.4.188:8121 lastseen=735.1hrs
ThreadRPCServer method=listaccounts
connection timeout
trying connection 92.241.33.169:8121 lastseen=296.9hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 166.137.136.127:8121 lastseen=429.4hrs
connection timeout
trying connection 67.183.83.146:8121 lastseen=716.5hrs
connection timeout
trying connection 2.248.90.41:8121 lastseen=136.2hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 201.56.8.130:8121 lastseen=201.8hrs
connection timeout
trying connection 120.188.81.78:8121 lastseen=209.5hrs
connection timeout
trying connection 67.183.83.146:8121 lastseen=716.5hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 107.205.21.78:8121 lastseen=586.4hrs
connection timeout
trying connection 99.166.78.109:8121 lastseen=235.4hrs
trying connection 103.253.43.162:8121 lastseen=384934.3hrs
connection timeout
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connect() failed after select(): Connection refused
trying connection 91.201.245.40:8121 lastseen=334.0hrs
trying connection 104.172.24.79:8121 lastseen=384934.3hrs
connection timeout
connection timeout
trying connection 47.217.29.246:8121 lastseen=789.1hrs
trying connection 115.28.42.60:8121 lastseen=384934.3hrs
connection timeout
connection timeout
trying connection 1.122.55.127:8121 lastseen=12581.7hrs
trying connection 137.110.85.246:8121 lastseen=384934.3hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
connection timeout
trying connection 178.219.162.94:8121 lastseen=736.6hrs
trying connection 137.74.104.135:8121 lastseen=384934.3hrs
connect() failed after select(): Connection refused
trying connection 139.5.68.111:8121 lastseen=384934.3hrs
connection timeout
trying connection 120.188.81.78:8121 lastseen=209.5hrs
connection timeout
trying connection 168.253.112.12:8121 lastseen=384934.3hrs
connect() failed after select(): Connection refused
trying connection 171.49.143.168:8121 lastseen=384934.3hrs
connection timeout
trying connection 27.0.87.66:8121 lastseen=4480.9hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
connection timeout
trying connection 173.208.211.58:8121 lastseen=384934.3hrs
Flushed 12376 addresses to peers.dat  36ms
connection timeout
trying connection 85.140.3.210:8121 lastseen=259.6hrs
connection timeout
trying connection 173.254.231.193:8121 lastseen=384934.3hrs
connect() failed after select(): Connection refused
connection timeout
trying connection 178.217.30.207:8121 lastseen=384934.3hrs
trying connection 188.158.247.144:8121 lastseen=155.5hrs
connection timeout
connection timeout
trying connection 181.228.29.105:8121 lastseen=384934.3hrs
trying connection 67.172.122.0:8121 lastseen=586.6hrs
getheaders 563496 to 3a3a8a5c1d8a38635ca7
received block 45398256ec2df49e12ee
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c0832fd  000000000832fd00000000000000000000000000000000000000000000000000
After:  1c08217e  0000000008217e00000000000000000000000000000000000000000000000000
Entered CheckSetBestChain: new best=45398256ec2df49e12ee  height=563497  work=60246097523642229  date=01/30/17 08:07:05
ProcessBlock: ACCEPTED
CreateNewBlock(): total size 1000
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c08217e  0000000008217e00000000000000000000000000000000000000000000000000
After:  1c081025  0000000008102500000000000000000000000000000000000000000000000000
ThreadRPCServer method=getblock
getheaders 563497 to 45398256ec2df49e12ee
getblocks -1 to 00000000000000000000 limit 500
getblocks -1 to 00000000000000000000 limit 500
Flushing wallet.dat
Flushed wallet.dat 29ms
getblocks -1 to 00000000000000000000 limit 500
getblocks -1 to 00000000000000000000 limit 500
connection timeout
connection timeout
trying connection 188.0.136.130:8121 lastseen=384934.3hrs
trying connection 169.159.79.217:8121 lastseen=481.5hrs
connect() failed after select(): Connection refused
trying connection 188.162.84.123:8121 lastseen=384934.3hrs
connection timeout
trying connection 84.164.121.104:8121 lastseen=738.4hrs
connection timeout
connect() failed after select(): No route to host
trying connection 188.162.84.27:8121 lastseen=384934.3hrs
trying connection 88.184.191.151:8121 lastseen=500.1hrs
getheaders 563497 to 45398256ec2df49e12ee
connection timeout
connection timeout
trying connection 188.162.84.96:8121 lastseen=384934.3hrs
trying connection 62.194.179.146:8121 lastseen=209.2hrs
connection timeout
connection timeout
trying connection 189.216.56.102:8121 lastseen=384934.3hrs
trying connection 95.25.4.188:8121 lastseen=735.1hrs
connection timeout
connection timeout
trying connection 192.124.224.244:8121 lastseen=384934.3hrs
trying connection 95.25.4.188:8121 lastseen=735.1hrs
connect() failed after select(): Connection refused
trying connection 192.99.62.23:8121 lastseen=384934.3hrs
getheaders 563497 to 45398256ec2df49e12ee
connect() failed after select(): No route to host
trying connection 2.248.90.41:8121 lastseen=384934.3hrs
connection timeout
trying connection 133.218.56.40:8121 lastseen=312.3hrs
connection timeout
trying connection 200.217.221.98:8121 lastseen=384934.3hrs
connection timeout
trying connection 107.183.242.76:8121 lastseen=159.7hrs
connect() failed after select(): Connection refused
trying connection 78.225.114.147:8121 lastseen=211.8hrs
connection timeout
trying connection 23.117.147.116:8121 lastseen=384934.3hrs
getheaders 563497 to 45398256ec2df49e12ee
connection timeout
trying connection 81.153.137.249:8121 lastseen=8862.7hrs
connection timeout
trying connection 24.247.150.25:8121 lastseen=384934.3hrs
connection timeout
trying connection 27.33.21.250:8121 lastseen=575.2hrs
connection timeout
trying connection 24.49.180.69:8121 lastseen=384934.3hrs
connected 24.49.180.69:8121
send version message: version 60018,
 blocks=563497, us=0.0.0.0:0, them=24.49.180.69:8121, peer=24.49.180.69:8121
trying connection 24.54.97.47:8121 lastseen=384934.3hrs
socket closed
disconnecting node 24.49.180.69:8121
ThreadRPCServer method=getinfo
keypool reserve 3
keypool return 3
connected 24.54.97.47:8121
send version message: version 60018, blocks=563497, us=0.0.0.0:0, them=24.54.97.47:8121, peer=24.54.97.47:8121
connection timeout
trying connection 35.46.78.142:8121 lastseen=384934.3hrs
trying connection 24.42.161.15:8121 lastseen=769.8hrs
socket closed
disconnecting node 24.54.97.47:8121
connect() failed after select(): No route to host
connect() failed after select(): Connection refused
trying connection 41.190.3.200:8121 lastseen=384934.3hrs
trying connection 24.223.137.160:8121 lastseen=791.9hrs
getheaders 563497 to 45398256ec2df49e12ee
connection timeout
connection timeout
trying connection 64.122.75.194:8121 lastseen=384934.3hrs
trying connection 85.26.186.165:8121 lastseen=4677.1hrs
connection timeout
connection timeout
trying connection 65.102.140.116:8121 lastseen=384934.3hrs
trying connection 83.221.170.27:8121 lastseen=32381.1hrs
connected 65.102.140.116:8121
send version message: version 60018, blocks=563497, us=0.0.0.0:0, them=65.102.140.116:8121, peer=65.102.140.116:8121
trying connection 67.181.49.30:8121 lastseen=384934.3hrs
socket closed
disconnecting node 65.102.140.116:8121
connection timeout
trying connection 73.132.239.109:8121 lastseen=10278.5hrs
getheaders 563497 to 45398256ec2df49e12ee
connection timeout
trying connection 70.81.199.49:8121 lastseen=384934.3hrs
connect() failed after select(): Connection refused
trying connection 72.51.230.44:8121 lastseen=384934.3hrs
ThreadRPCServer method=submitblock
ThreadRPCServer method=submitblock
Local has found possible valid block... queueing until timestamp is valid 
connection timeout
trying connection 149.3.96.233:8121 lastseen=435.0hrs
connection timeout
trying connection 73.185.207.88:8121 lastseen=384934.4hrs
connection timeout
trying connection 188.162.84.96:8121 lastseen=217.0hrs
Flushed 12376 addresses to peers.dat  455ms
connection timeout
trying connection 73.214.14.51:8121 lastseen=384934.4hrs
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c08217e  0000000008217e00000000000000000000000000000000000000000000000000
After:  1c081025  0000000008102500000000000000000000000000000000000000000000000000
Entered CheckSetBestChain: new best=bc43afbee5429e8e49ac  height=563498  work=60246233887628592  date=01/30/17 08:08:37
ProcessBlock: ACCEPTED
ERROR: ProcessBlock() : already have block 563498 bc43afbee5429e8e49ac
received block bc43afbee5429e8e49ac
ERROR: ProcessBlock() : already have block 563498 bc43afbee5429e8e49ac
received block bc43afbee5429e8e49ac
ERROR: ProcessBlock() : already have block 563498 bc43afbee5429e8e49ac
CreateNewBlock(): total size 1000
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c081025  0000000008102500000000000000000000000000000000000000000000000000
After:  1c07fef1  0000000007fef100000000000000000000000000000000000000000000000000
ThreadRPCServer method=getblock
getheaders 563498 to bc43afbee5429e8e49ac
received block c9236c8b19e94ceee5f5
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c08217e  0000000008217e00000000000000000000000000000000000000000000000000
After:  1c081025  0000000008102500000000000000000000000000000000000000000000000000
Entered CheckProcessBlock: ACCEPTED
getblocks -1 to 00000000000000000000 limit 500
getblocks -1 to 00000000000000000000 limit 500
connection timeout
trying connection 189.15.214.246:8121 lastseen=214.4hrs
Flushing wallet.dat
Flushed wallet.dat 24ms
connect() failed after select(): Connection refused
trying connection 173.22.81.221:8121 lastseen=707.8hrs
connection timeout
trying connection 77.37.166.206:8121 lastseen=384934.4hrs
connection timeout
trying connection 189.15.214.246:8121 lastseen=214.4hrs
connection timeout
trying connection 80.86.85.65:8121 lastseen=384934.4hrs
connect() failed after select(): Connection refused
connect() failed after select(): Connection refused
trying connection 178.34.59.59:8121 lastseen=231.3hrs
trying connection 85.140.3.210:8121 lastseen=384934.4hrs
getheaders 563498 to bc43afbee5429e8e49ac
connection timeout
connection timeout
trying connection 45.37.13.248:8121 lastseen=665.8hrs
trying connection 88.206.141.110:8121 lastseen=384934.4hrs
connect() failed after select(): No route to host
trying connection 88.206.186.58:8121 lastseen=384934.4hrs
connected 88.206.186.58:8121
send version message: version 60018, blocks=563498, us=0.0.0.0:0, them=88.206.186.58:8121, peer=88.206.186.58:8121
trying connection 91.121.6.41:8121 lastseen=384934.4hrs
connected 91.121.6.41:8121
send version message: version 60018, blocks=563498, us=0.0.0.0:0, them=91.121.6.41:8121, peer=91.121.6.41:8121
trying connection 91.134.233.254:8121 lastseen=384934.4hrs
connection timeout
socket closed
disconnecting node 91.121.6.41:8121
trying connection 133.218.56.40:8121 lastseen=312.3hrs
connection timeout
trying connection 91.201.245.200:8121 lastseen=384934.4hrs
connection timeout
trying connection 192.162.0.100:8121 lastseen=686.0hrs
getheaders 563498 to bc43afbee5429e8e49ac
connection timeout
trying connection 92.3.45.1:8121 lastseen=384934.4hrs
connection timeout
trying connection 206.248.191.18:8121 lastseen=27517.6hrs
connection timeout
trying connection 93.139.160.78:8121 lastseen=384934.4hrs
connection timeout
trying connection 88.71.38.196:8121 lastseen=358.5hrs
connect() failed after select(): Connection refused
received block 6e242369cafc73372117
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c081025  0000000008102500000000000000000000000000000000000000000000000000
After:  1c07fef1  0000000007fef100000000000000000000000000000000000000000000000000
Entered CheckREORGANIZE
REORGANIZE: Disconnect 1 blocks; 45398256ec2df49e12ee..bc43afbee5429e8e49ac
REORGANIZE: Connect 2 blocks; 45398256ec2df49e12ee..6e242369cafc73372117
REORGANIZE: done
connect() failed after select(): No route to host
SetBestChain: new best=6e242369cafc73372117  height=563499  work=60246371397659827  date=01/30/17 08:09:16
ProcessBlock: ACCEPTED
CreateNewBlock(): total size 1000
  Median Time between blocks is: 93 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c07fef1  0000000007fef100000000000000000000000000000000000000000000000000
After:  1c07ede2  0000000007ede200000000000000000000000000000000000000000000000000
ThreadRPCServer method=getblock
getheaders 563499 to 6e242369cafc73372117
getblocks -1 to 00000000000000000000 limit 500
trying connection 78.150.159.68:8121 lastseen=304.4hrs
getblocks -1 to 00000000000000000000 limit 500
Flushing wallet.dat
Flushed wallet.dat 17ms
connection timeout
trying connection 213.163.50.173:8121 lastseen=435.4hrs
connect() failed after select(): Connection refused
trying connection 188.162.64.24:8121 lastseen=634.3hrs
ThreadRPCServer method=submitblock
Local has found possible valid block... queueing until timestamp is valid 
connection timeout
trying connection 176.74.124.139:8121 lastseen=256.6hrs
connect() failed after select(): No route to host
trying connection 78.85.73.222:8121 lastseen=448.5hrs
ThreadRPCServer method=submitblock
connection timeout
trying connection 104.238.169.155:8121 lastseen=203.5hrs
connect() failed after select(): Connection refused
trying connection 93.142.42.1:8121 lastseen=331.8hrs
connection timeout
trying connection 94.64.125.214:8121 lastseen=389.7hrs
connection timeout
trying connection 85.140.3.210:8121 lastseen=259.6hrs
connection timeout
trying connection 124.187.49.151:8121 lastseen=472.0hrs
socket no message in first 60 seconds, 0 1
disconnecting node 88.206.186.58:8121
connection timeout
trying connection 78.85.73.222:8121 lastseen=448.5hrs
ThreadRPCServer method=submitblock
ThreadRPCServer method=submitblock
connection timeout
trying connection 78.150.159.213:8121 lastseen=202.5hrs
ThreadRPCServer method=submitblock
connection timeout
trying connection 2.248.90.41:8121 lastseen=136.3hrs
ThreadRPCServer method=submitblock
Flushed 12376 addresses to peers.dat  44ms
connection timeout
trying connection 77.250.227.41:8121 lastseen=157.2hrs
connection timeout
trying connection 5.153.234.82:8121 lastseen=856.1hrs
connect() failed after select(): Connection refused
trying connection 201.56.8.130:8121 lastseen=201.9hrs
connection timeout
trying connection 95.70.99.249:8121 lastseen=659.8hrs
connection timeout
trying connection 189.15.214.246:8121 lastseen=214.4hrs
connect() failed after select(): Connection refused
trying connection 46.129.108.70:8121 lastseen=26419.8hrs
ThreadRPCServer method=submitblock
ThreadRPCServer method=submitblock
connection timeout
trying connection 131.161.30.5:8121 lastseen=2490.1hrs
connect() failed after select(): No route to host
trying connection 201.190.18.190:8121 lastseen=4661.0hrs
ThreadRPCServer method=submitblock
connection timeout
trying connection 84.189.2.229:8121 lastseen=3100.6hrs
ThreadRPCServer
 method=submitblock
connection timeout
trying connection 77.179.93.90:8121 lastseen=290.5hrs
connect() failed after select(): No route to host
trying connection 122.168.15.35:8121 lastseen=213.1hrs
connection timeout
trying connection 188.18.108.10:8121 lastseen=732.8hrs
connection timeout
trying connection 190.105.155.194:8121 lastseen=303.6hrs
connect() failed after select(): Connection refused
trying connection 120.188.81.78:8121 lastseen=209.6hrs
connection timeout
trying connection 177.220.226.34:8121 lastseen=464.7hrs
trying connection 103.253.43.162:8121 lastseen=384934.4hrs
connect() failed after select(): Connection refused
trying connection 189.216.56.38:8121 lastseen=573.1hrs
connect() failed after select(): Connection refused
trying connection 104.172.24.79:8121 lastseen=384934.4hrs
ThreadRPCServer method=submitblock
ThreadRPCServer method=submitblock
connection timeout
trying connection 118.209.74.12:8121 lastseen=403.8hrs
connection timeout
trying connection 115.28.42.60:8121 lastseen=384934.4hrs
ThreadRPCServer method=submitblock
connection timeout
trying connection 177.204.9.72:8121 lastseen=897.8hrs
connection timeout
trying connection 137.110.85.246:8121 lastseen=384934.4hrs
connection timeout
ThreadRPCServer method=submitblock
trying connection 79.126.186.231:8121 lastseen=833.8hrs
connection timeout
trying connection 137.74.104.135:8121 lastseen=384934.4hrs
connect() failed after select(): Connection refused
trying connection 139.5.68.111:8121 lastseen=384934.4hrs
connection timeout
trying connection 92.113.97.177:8121 lastseen=614.3hrs
connection timeout
trying connection 168.253.112.12:8121 lastseen=384934.4hrs
connect() failed after select(): Connection refused
connection timeout
trying connection 171.49.143.168:8121 lastseen=384934.4hrs
trying connection 105.112.42.88:8121 lastseen=339.7hrs
connection timeout
connection timeout
trying connection 173.208.211.58:8121 lastseen=384934.4hrs
trying connection 87.157.67.103:8121 lastseen=471.6hrs
Flushed 12376 addresses to peers.dat  50ms
connection timeout
connection timeout
trying connection 173.254.231.193:8121 lastseen=384934.4hrs
trying connection 197.211.63.54:8121 lastseen=600.4hrs
connect() failed after select(): Connection refused
trying connection 178.217.30.207:8121 lastseen=384934.4hrs
connection timeout

**Comment by microguy** [comment]
Akumaburn stays relatively busy during the week and might not have an opportunity to revisit this matter again until Sunday. We welcome and encourage others to take a deeper look in the meantime and weigh in with suggestions, recommendations, and/or potential solutions.

**Comment by p3yot3** [comment]
The freezing problem remains with the latest version 0.7.5

**Comment by Stouse49** [comment]
IconFirm, which version of P2Pool are you using?  This link shows that any version above 14.0 will not work since it requires Bitcoin 0.8.5 or higher code.  GoldCoin's code base is from 0.6.9.  https://github.com/p2pool/p2pool/tree/14.0

Your logs that you posted contain this line many times:
"ThreadRPCServer method=submitblock"

This line is showing the RPC method that is being called.  GoldCoin does not support that method ("submitblock").

This line also occurred once:
"Local has found possible valid block... queueing until timestamp is valid"

If this line is present, then in certain cases, the daemon will sleep for a period of time, until certain requirements are met with the time of the solved block and the current adjusted time.

**Comment by Stouse49** [comment]
p3yot3, what is in your GoldCoin logs?

**Comment by p3yot3** [comment]
@Stouse49  Sorry, I deleted everything in a fit of depression as soon as I saw the "new" daemon freezing again....lol
I will recompile & try running it again when I have the time & patience then post my output. One thing I do know though - it is not a p2pool problem, it is a Goldcoin problem. I tried it with p2pool v13 & v14 with the same results.

**Comment by Stouse49** [comment]
I tried to set up a p2pool on a linux system.  I had errors from not configuring the p2pool correctly, probably, so I couldn't replicate the problem.

**Comment by p3yot3** [comment]
Still experiencing freezing up issue:

2017-02-19 15:50:56.861630 > ########################################
2017-02-19 15:50:56.861757 > >>> Warning: LOST CONTACT WITH BITCOIND for 7.7 minutes! Check that it isn't frozen or dead!
2017-02-19 15:50:56.861802 > ########################################
2017-02-19 15:50:59.868550 > ########################################
2017-02-19 15:50:59.868769 > >>> Warning: LOST CONTACT WITH BITCOIND for 7.7 minutes! Check that it isn't frozen or dead!
2017-02-19 15:50:59.868874 > ########################################
2017-02-19 15:51:00.363465 Punishing share for 'Block-stale detected! height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) < height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) or 1c0574d6 != 1c056931'! Jumping from d321691d to 100ff89c!
2017-02-19 15:51:00.368743 Punishing share for 'Block-stale detected! height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) < height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) or 1c0574d6 != 1c056931'! Jumping from d321691d to 100ff89c!
2017-02-19 15:51:02.599820 > Unhandled error in Deferred:
2017-02-19 15:51:02.600148 > Unhandled Error
2017-02-19 15:51:02.600238 > Traceback (most recent call last):
2017-02-19 15:51:02.600319 > Failure: twisted.internet.defer.TimeoutError: Getting http://127.0.0.1:8122/ took longer than 30 seconds.
2017-02-19 15:51:02.602177 > Unhandled error in Deferred:
2017-02-19 15:51:02.602588 > Unhandled Error
2017-02-19 15:51:02.602699 > Traceback (most recent call last):
2017-02-19 15:51:02.602771 > Failure: twisted.internet.defer.TimeoutError: Getting http://127.0.0.1:8122/ took longer than 30 seconds.
2017-02-19 15:51:02.871977 > ########################################
2017-02-19 15:51:02.872185 > >>> Warning: LOST CONTACT WITH BITCOIND for 7.8 minutes! Check that it isn't frozen or dead!
2017-02-19 15:51:02.872235 > ########################################
2017-02-19 15:51:03.130552 Got new merged mining work!
2017-02-19 15:51:05.379615 Punishing share for 'Block-stale detected! height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) < height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) or 1c0574d6 != 1c056931'! Jumping from d321691d to 100ff89c!
2017-02-19 15:51:05.384682 Punishing share for 'Block-stale detected! height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) < height(ee78d5fd8691dcffd51d05e87ea5f6edcacd5494a2164c9facfddbf9be2cd38c) or 1c0574d6 != 1c056931'! Jumping from d321691d to 100ff89c!
2017-02-19 15:51:05.879252 > ########################################
2017-02-19 15:51:05.879471 > >>> Warning: LOST CONTACT WITH BITCOIND for 7.8 minutes! Check that it isn't frozen or dead!

..sometimes up to 10 minutes at a time.

**Comment by microguy** [comment]
The Goldcoin client is currently undergoing a code rebase. I appreciate you updating us and providing this debug information, but making accusations like we're ignoring the problem are not accurate or constructive.

**Comment by Stouse49** [comment]
The log that would be helpful is the debug.log from ~/.goldcoin.

**Comment by p3yot3** [comment]
@Stouse49  Here you go:

received block 9f7b57ed468810c1089c
ProcessBlock() : Time between last 5 blocks is: 600  Median Time between blocks is: 100 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 123 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c079414  0000000007941400000000000000000000000000000000000000000000000000
After:  1c07a440  0000000007a44000000000000000000000000000000000000000000000000000
SetBestChain: new best=9f7b57ed468810c1089c  height=576871  work=62241672343392946  date=02/21/17 22:05:18
ProcessBlock: ACCEPTED
CreateNewBlock(): total size 1000
  Median Time between blocks is: 100 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c07a440  0000000007a44000000000000000000000000000000000000000000000000000
After:  1c07b48c  0000000007b48c00000000000000000000000000000000000000000000000000
getblocks -1 to 00000000000000000000 limit 500
ThreadRPCServer method=getblock
getheaders 576871 to 9f7b57ed468810c1089c
getblocks -1 to 00000000000000000000 limit 500
getblocks -1 to 00000000000000000000 limit 500
Flushing wallet.dat
Flushed wallet.dat 20ms
connection timeout
getblocks -1 to 00000000000000000000 limit 500
trying connection 166.137.136.66:8121 lastseen=102.3hrs
connection timeout
trying connection 91.236.254.71:8121 lastseen=150.0hrs
connection timeout
trying connection 166.137.136.85:8121 lastseen=7.1hrs
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
trying connection 74.130.200.135:8121 lastseen=347.6hrs
connection timeout
trying connection 178.115.129.242:8121 lastseen=3107.4hrs
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
trying connection 180.245.132.124:8121 lastseen=625.5hrs
connection timeout
trying connection 94.25.229.232:8121 lastseen=77.9hrs
connection timeout
trying connection 211.24.6.216:8121 lastseen=26866.7hrs
Flushed 13823 addresses to peers.dat  44ms
getheaders 576871 to 9f7b57ed468810c1089c
trying connection 192.168.1.50:8121 lastseen=385476.3hrs
connection timeout
trying connection 130.180.212.19:8121 lastseen=711.2hrs
connect() failed after select(): No route to host
trying connection 104.185.96.131:8121 lastseen=385476.3hrs
connection timeout
trying connection 184.164.129.202:8121 lastseen=4.6hrs
connection timeout
trying connection 107.185.176.131:8121 lastseen=385476.3hrs
connection timeout
trying connection 193.154.26.132:8121 lastseen=417.4hrs
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
trying connection 108.61.10.90:8121 lastseen=385476.3hrs
connected 108.61.10.90:8121
send version message: version 60020, blocks=576871, us=92.3.36.155:8121, them=108.61.10.90:8121, peer=108.61.10.90:8121
socket closed
disconnecting node 108.61.10.90:8121
trying connection 163.172.135.134:8121 lastseen=385476.3hrs
connected 163.172.135.134:8121
send version message: version 60020, blocks=576871, us=92.3.36.155:8121, them=163.172.135.134:8121, peer=163.172.135.134:8121
socket closed
disconnecting node 163.172.135.134:8121
trying connection 217.175.119.125:8121 lastseen=385476.3hrs
ThreadRPCServer method=getblock
connection timeout
trying connection 107.205.21.78:8121 lastseen=190.6hrs
connection timeout
trying connection 68.229.185.16:8121 lastseen=385476.3hrs
connection timeout
trying connection 122.175.215.101:8121 lastseen=279.5hrs
connection timeout
trying connection 68.55.199.104:8121 lastseen=385476.3hrs
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
trying connection 167.62.22.19:8121 lastseen=199.2hrs
connect() failed after select(): Connection refused
trying connection 24.246.28.36:8121 lastseen=14.9hrs
connect() failed after select(): Connection refused
trying connection 93.139.182.25:8121 lastseen=137.2hrs
connection timeout
connect() failed after select(): Connection refused
trying connection 78.66.83.88:8121 lastseen=385476.3hrs
trying connection 85.140.0.126:8121 lastseen=110.5hrs
connection timeout
connection timeout
trying connection 81.153.89.196:8121 lastseen=385476.3hrs
trying connection 96.235.143.102:8121 lastseen=2.0hrs
connection timeout
connection timeout
trying connection 88.99.68.228:8121 lastseen=385476.3hrs
trying connection 212.96.122.38:8121 lastseen=377.7hrs
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
connection timeout
trying connection 94.25.227.219:8121 lastseen=385476.3hrs
trying connection 100.92.89.113:8121 lastseen=222.9hrs
connection timeout
connection timeout
trying connection 94.25.229.232:8121 lastseen=385476.3hrs
trying connection 178.76.241.58:8121 lastseen=28503.9hrs
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
connection timeout
trying connection 96.235.143.102:8121 lastseen=385476.3hrs
trying connection 64.222.216.7:8121 lastseen=118.7hrs
connection timeout
connection timeout
trying connection 81.153.89.168:8121 lastseen=289.9hrs
connection timeout
trying connection 216.232.236.149:8121 lastseen=3.6hrs
ThreadRPCServer method=getblock
getheaders 576871 to 9f7b57ed468810c1089c
connection timeout
trying connection 189.216.56.102:8121 lastseen=158.3hrs
connection timeout
trying connection 5.187.68.135:8121 lastseen=5.6hrs
ThreadRPCServer method=submitblock
Local has found possible valid block... queueing until timestamp is valid 
connection timeout
trying connection 73.109.40.66:8121 lastseen=306.6hrs
connection timeout
trying connection 5.187.68.135:8121 lastseen=5.6hrs
Flushed 13823 addresses to peers.dat  53ms
connection timeout
trying connection 100.66.208.130:8121 lastseen=9155.1hrs
connection timeout
trying connection 46.72.74.187:8121 lastseen=14.5hrs
connection timeout
trying connection 80.12.94.197:8121 lastseen=40.1hrs
connection timeout
trying connection 203.111.232.6:8121 lastseen=398.2hrs
connection timeout
trying connection 24.246.28.36:8121 lastseen=15.0hrs
connect() failed after select(): Connection refused
trying connection 71.77.128.232:8121 lastseen=5.5hrs
connection timeout
ThreadRPCServer method=submitblock
trying connection 87.17.120.121:8121 lastseen=173.0hrs
connection timeout
trying connection 173.81.58.185:8121 lastseen=441.1hrs
connection timeout
trying connection 216.71.205.204:8121 lastseen=2.6hrs
connection timeout
trying connection 46.195.128.59:8121 lastseen=270.9hrs
connect() failed after select(): Connection refused
trying connection 87.117.38.94:8121 lastseen=49.7hrs
connection timeout
trying connection 85.140.3.210:8121 lastseen=809.7hrs
connection timeout
trying connection 173.70.28.225:8121 lastseen=700.2hrs
connection timeout
trying connection 174.61.115.71:8121 lastseen=252.1hrs
ThreadRPCServer method=submitblock
connection timeout
trying connection 104.238.169.69:8121 lastseen=100.1hrs
connect() failed after select(): Connection refused
trying connection 93.44.189.60:8121 lastseen=3.6hrs
connection timeout
trying connection 89.151.188.217:8121 lastseen=368.9hrs
connection timeout
trying connection 37.233.55.114:8121 lastseen=23257.4hrs
connection timeout
trying connection 217.72.97.23:8121 lastseen=727.1hrs
trying connection 192.168.1.50:8121 lastseen=385476.4hrs
connect() failed after select(): No route to host
trying connection 104.185.96.131:8121 lastseen=385476.4hrs
connection timeout
trying connection 83.29.219.104:8121 lastseen=12764.0hrs
connection timeout
trying connection 107.185.176.131:8121 lastseen=385476.4hrs
connection timeout
trying connection 85.140.80.40:8121 lastseen=157.2hrs
Flushed 13823 addresses to peers.dat  49ms
connection timeout
trying connection 108.61.10.90:8121 lastseen=385476.4hrs
connected 108.61.10.90:8121
send version message: version 60020, blocks=576871, us=92.3.36.155:8121, them=108.61.10.90:8121, peer=108.61.10.90:8121
socket closed
disconnecting node 108.61.10.90:8121
trying connection 163.172.135.134:8121 lastseen=385476.4hrs
connected 163.172.135.134:8121
send version message: version 60020, blocks=576871, us=92.3.36.155:8121, them=163.172.135.134:8121, peer=163.172.135.134:8121
socket closed
disconnecting node 163.172.135.134:8121
trying connection 217.175.119.125:8121 lastseen=385476.4hrs
connection timeout
trying connection 190.219.207.39:8121 lastseen=3879.1hrs
connection timeout
trying connection 68.229.185.16:8121 lastseen=385476.4hrs
connection timeout
ThreadRPCServer method=submitblock
trying connection 193.154.26.132:8121 lastseen=417.4hrs
connection timeout
trying connection 68.55.199.104:8121 lastseen=385476.4hrs
connection timeout
trying connection 73.85.200.140:8121 lastseen=431.4hrs
connection timeout
trying connection 78.66.83.88:8121 lastseen=385476.4hrs
connection timeout
trying connection 213.238.88.74:8121 lastseen=4.1hrs
connection timeout
trying connection 81.153.89.196:8121 lastseen=385476.4hrs
connection timeout
trying connection 98.180.49.44:8121 lastseen=28473.6hrs
connection timeout
trying connection 88.99.68.228:8121 lastseen=385476.4hrs
connection timeout
trying connection 94.25.227.219:8121 lastseen=59.0hrs
connection timeout
trying connection 94.25.227.219:8121 lastseen=385476.4hrs
connection timeout
trying connection 188.99.81.81:8121 lastseen=391.3hrs
connect() failed after select(): No route to host
trying connection 88.206.141.110:8121 lastseen=11.4hrs
connect() failed after select(): No route to host
trying connection 87.157.64.228:8121 lastseen=532.1hrs
connection timeout
trying connection 94.25.229.232:8121 lastseen=385476.4hrs
connection timeout
trying connection 110.174.193.68:8121 lastseen=116.3hrs
connection timeout
trying connection 96.235.143.102:8121 lastseen=385476.4hrs
connection timeout
ThreadRPCServer method=submitblock
trying connection 41.224.102.70:8121 lastseen=1320.2hrs
connection timeout
connection timeout
trying connection 88.206.141.110:8121 lastseen=11.4hrs
connect() failed after select(): No route to host
trying connection 84.164.125.224:8121 lastseen=462.3hrs
connection timeout
trying connection 196.210.63.130:8121 lastseen=85.5hrs
connection timeout
ProcessBlock() : Time between last 5 blocks is: 600  Median Time between blocks is: 100 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c07a440  0000000007a44000000000000000000000000000000000000000000000000000
After:  1c07b48c  0000000007b48c00000000000000000000000000000000000000000000000000
SetBestChain: new best=8b2aca89202ee7bd20e1  height=576872  work=62241815039609841  date=02/21/17 22:10:30
ProcessBlock: ACCEPTED
received block 8b2aca89202ee7bd20e1
ERROR: ProcessBlock() : already have block 576872 8b2aca89202ee7bd20e1
trying connection 96.19.143.29:8121 lastseen=390.7hrs
CreateNewBlock(): total size 1000
  Median Time between blocks is: 100 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 
GetNextWorkRequired RETARGET
nTargetTimespan = 7200    nActualTimespan = 7020
Before: 1c07b48c  0000000007b48c00000000000000000000000000000000000000000000000000
After:  1c07a41d  0000000007a41d00000000000000000000000000000000000000000000000000
ERROR: ProcessBlock() : already have block 576872 8b2aca89202ee7bd20e1
ERROR: ProcessBlock() : already have block 576872 8b2aca89202ee7bd20e1
ERROR: ProcessBlock() : already have block 576872 8b2aca89202ee7bd20e1
ERROR: ProcessBlock() : already have block 576872 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
ThreadRPCServer method=getblock
received getdata for: block 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
connect() failed after select(): Connection refused
getheaders 576872 to 8b2aca89202ee7bd20e1
received getdata for: block 8b2aca89202ee7bd20e1
trying connection 84.164.103.49:8121 lastseen=50.6hrs
Flushing wallet.dat
Flushed wallet.dat 14ms
received block ff277aa1b27b1e8349cd
ProcessBlock() : Time between last 5 blocks is: 600  Median Time between blocks is: 100 
 GetNextWorkRequired(): May Fork mode 
 GetNextWorkRequired(): Average time between blocks over the last 120 blocks is: 122 

i tried to take this log while the daemon had frozen, but as there's no time stamp on the logs it's hard to marry them up with the p2pool logs.

**Comment by Stouse49** [comment]
Both users have reported their logs which both contain this line "Local has found possible valid block... queueing until timestamp is valid"

main.cpp Line 2347 may be part of the code that is causing this issue:

```
printf("Local has found possible valid block... queueing until timestamp is valid \n");
					//Since we know that GetBlockTime() is greater than GetAdjustedTime()
					//We sleep the difference
					//Doesn't hurt to check twice
					if (GetBlockTime() > GetAdjustedTime() + 45) {
						//Set a variable prior to sleeping warning other connectionhandler not to sleep as well.
						dontWaitForBlockStamp = true;
						Sleep(1000 * (GetBlockTime() - (GetAdjustedTime() + 45)));
					}
```

Under certain circumstances, where a block time is more than 45 seconds ahead of the adjusted time, the thread will sleep for 45 seconds more than the difference between the adjusted time and the block time.

I think this is the cause of the "Warning: LOST CONTACT WITH BITCOIND for 7.8 minutes! Check that it isn't frozen or dead!" message from p2pool.  When the thread that responds to RPC commands is sleeping, then it won't respond to any other RPC requests.

**Comment by Stouse49** [comment]
I reproduced the error on my system by setting my clock to 20 minutes ahead.  GoldCoin requires that the computer time be 45 seconds within the Network time.

**Comment by p3yot3** [comment]
I'm aware of the importance of time settings, especially for GLD, so all my equipment is set to use the same time servers & poll them frequently to ensure accuracy. I'm pretty sure that changing your clock by as much as 20 minutes would mess a whole lot of stuff up.....
In order to accurately trace the problem, may I suggest that the debug logs be timestamped as standard, the same as all other coin daemon logs. This would help troubleshooting immensely.
I've also noticed that the time recorded for mined blocks is the the time the payment arrived in my wallet - not the time it was mined. This means that if I turn off my recieving wallet (which I do every night) then turn it on again later (the next morning), all mined payments show the time they arrived in my wallet - when they should instead show the time the block payments were mined. There is also no payment address listed, just an N/A - this shouldn't be the case.
I'm not suggesting that any or all of these issues are the cause of the freezing problem, (even though they could be a contributory factor), but that they are teething problems that need to be eradicated to ensure that they are not.

**Comment by Stouse49** [comment]
As an engineer it is often useful to test extreme cases (as in the case of a clock that is 20 minutes ahead) to see what kinds of problems or trends emerge.  

Since your clocks are set correctly, then I am at a loss to explain the reason for the freezing problem, but I still think the code I noted above is the part that is causing the freezing given the message in the log.  Granted there was only one such message in the log, but that code is executed when a block comes thru from another node and is validated.

Indeed, time stamps would be very helpful in debugging issues.  My android wallet does put time stamps in the log, which has helped solve many issues.

I have noticed that the times recorded for transactions are is the time that the client receives them, not the time of the block they were first confirmed in.  This is by design in the code (in wallet.cpp).  Perhaps newer version of Bitcoin Core have change this to the block time.

Regarding mined transactions showing with "N/A" for the address, it appears to be by design in the code to display "n/a" for mined or generated transactions (qt/transactionrecord.cpp).  It has been that way since the beginning of GoldCoin according to my transaction history.  Newer versions of Bitcoin Core may also correct this issue.

**Comment by p3yot3** [comment]
That's it? You can't find a solution to a problem so you close it & hope nobody notices & it will go away?

Is that how development works with GLD?

**Comment by d0gZpAw** [comment]
After reading the comments by @Stouse49, I decided to test some of the speculations in the thread.

I am currently running two Goldcoin p2pool-v14.0.11 nodes on different VPS providers, one west coast, one east coast.
Since there are no timestamps in Goldcoin's debug.log, I ran the following command several times in a row concurrently, while monitoring the status pages for both p2pool nodes;

`tail -f .goldcoin/debug.log | grep -m 1 "queueing"`

The first time this showed up in the log, the daemon on both nodes would not respond to 'getinfo' and showed the "LOST CONTACT" error until, I assume, the block timestamp caught up.

However, I ran the same tail command again, and a few minutes later, I noticed p2pool was showing the "LOST CONTACT" notice, and the daemon RPC was again unresponsive to getinfo.  The debug.log on both nodes do not match for this period, as one node was processing a block the other node had found and broadcasted.  The queueing message did not show up in the logs during this time, but p2pool was broadcasting a block.

I ran this test several more times over the course of an hour, and it seems the RPC connection issues alternate between queueing and something else, but both seem to be related to broadcasting blocks.

I also tried p2pool-v16 (reverting commit https://github.com/p2pool/p2pool/commit/ce9e5a2b5546a7adac8b686502d78c75ca414e06 to fix hash > target error), but unfortunately it seems something must have changed in bitcoin and litecoin since v14, and mining Goldcoin results in many DOA errors and much reduced hash rates.

In conclusion, I believe queueing is responsible for many of the "LOST CONTACT" notices in p2pool, but not all of them.  Whether this RPC interruption has any impact on miner income is difficult to quantify without a reliable data set, but it does not seem to increase the orphan rate, as many blocks that queue are also valid.

**Comment by crusher72** [comment]
I am losing contact with the coin daemon, but also I am getting Incomplete Block found, method not found 

error when requesting a non cached value.
error when requesting a non cached value.

Over and over again until it just freezes. 

I have tried a lot of p2p versions, 14.00 is the only one that works. But I can only run it for about a minute before it freezes.
